test/test_channel.lua

```diff
@@ -0,0 +1,402 @@
+local test = require 'regress'
+local ipc = require 'libipc'
+
+test {
+   -- it should be possible to open a channel, write something to it
+   -- and read it back within the same thread.
+   openReadWriteSameThread = function()
+      local c = ipc.channel()
+      local data = 10
+      local status = c:write(data)
+      test.mustBeTrue(status == ":open")
```

Comment:
"instead of status == ":open" it would be status == ipc.channel.OPEN. I'm not sure if that is what you were suggesting in the first place."

Yes, just an enum.